src/php/Command/FormatCommand.php

```diff
@@ -15,24 +17,43 @@ final class FormatCommand
     private FormatterInterface $formatter;
     private CommandIoInterface $io;
     private PathFilterInterface $pathFilter;
+    private ExceptionPrinterInterface $exceptionPrinter;
 
     /** @var list<string> */
     private array $successfulFormattedFilePaths = [];
 
     public function __construct(
         FormatterInterface $formatter,
         CommandIoInterface $io,
-        PathFilterInterface $pathFilter
+        PathFilterInterface $pathFilter,
+        ExceptionPrinterInterface $exceptionPrinter
     ) {
         $this->formatter = $formatter;
         $this->io = $io;
         $this->pathFilter = $pathFilter;
+        $this->exceptionPrinter = $exceptionPrinter;
     }
 
     /**
      * @param list<string> $paths
      */
     public function run(array $paths): void
+    {
+        try {
+            $this->formatPaths($paths);
+        } catch (ParserException $e) {
```

Comment:
Isn't it better to handle the exception just around ` $this->formatter->formatFile($path);`. Then all other files can still be formatted?

Reply:
Oh, clever, I didn't see that. I will try it 👌🏻